tools/create_testgrid_tickets.py

```diff
@@ -0,0 +1,161 @@
+
+*Test Status*: {status}
+*Test Details:* [{test_id}|https://testgrid.k8s.io/redhat-assisted-installer#{test_id}]
+*Last Test failure:* https://prow.ci.openshift.org/view/gcs/origin-ci-test/logs/{test_id}/{last_fail}
+{tests_list}
+"""
+
+
+def get_credentials_from_netrc(server, netrc_file=DEFAULT_NETRC_FILE):
+    cred = netrc.netrc(os.path.expanduser(netrc_file))
+    username, _, password = cred.authenticators(server)
+    return username, password
+
+
+def get_jira_client(username, password):
+    logger.info("log-in with username: %s", username)
+    return jira.JIRA(JIRA_SERVER, basic_auth=(username, password))
+
+
+def format_summary(failure_data):
+    return JIRA_SUMMARY.format(**failure_data)
+
+def get_all_triage_tickets(jclient):
+    query = 'filter in ("assisted installer triaging CI issues")'
+    return [i.fields.summary for i in jclient.search_issues(query, maxResults=None)]
+
+def add_watchers(jclient, issue):
+    for watcher in DEFAULT_WATCHERS:
+        jclient.add_watcher(issue.key, watcher)
+
+def get_last_failure_instance(test_id):
+    res = requests.get(TEST_RUNS_URL.format(test_id))
+    res.raise_for_status()
+    j = res.json()
+    return j['changelists'][0]
+
+def get_last_pass(test_data):
+    return [t for t in test_data['tests']
+            if t['display_name'] == 'Overall'][0]['pass_timestamp']
+
+def format_labels(failure_data):
+    return ["no-qe",
+            "AI_CI_TRIAGE",
+            "AI_CI_TEST_{test_id}".format(**failure_data)]
+
+def format_description(failure_data):
+    tests_list = "*Failed tests:*\n"
+    for test in failure_data['failed_tests']:
+        tests_list += " * {}\n".format(test)
+    failure_data["tests_list"] = tests_list
+    return JIRA_DESCRIPTION.format(**failure_data)
+
+def create_jira_ticket(jclient, existing_tickets, test_id, test_data):
+    last_pass_timestamp = get_last_pass(test_data)
+    summary = format_summary({"test_id":test_id,
+                              "last_pass": last_pass_timestamp})
+    if summary in existing_tickets:
+        logger.debug("issue found: %s", summary)
+        return None
+
+    failed_tests = [t['display_name'] for t in test_data['tests'] if t['display_name'] != 'Overall']
+    test_status = test_data['status']
+    i = get_last_failure_instance(test_id)
+    new_issue = jclient.create_issue(project="MGMT",
+                                     description=format_description({"test_id": test_id,
+                                                                     "last_fail": i,
+                                                                     "failed_tests": failed_tests,
+                                                                     "status": test_status,
+                                                                    }),
+                                     summary=summary,
+                                     versions=[{'name': 'OpenShift {}'.format(TEST_VERSION_REGEX.findall(test_id)[0])}],
```

Comment:
If we either way only want one result:
```suggestion
                                     versions=[{'name': 'OpenShift {}'.format(TEST_VERSION_REGEX.search(test_id))}],
```

Reply:
This does not work cause search returns an object